### DISCUSSION TITLE
How to stop collector after onMessage function?

### DISCUSSION THREAD

**Discussion by amenbros** [IDjinn/Discord.js-Collector#48] [Q&A] [answered]
I've been scratching my heads for hours and I can't figure it out.

I want to stop the MessageCollector after receiving a message on 'onMessage'.

```
onMessage: async (controller, message) => {
                        return await message.reply('✅ | Success!').then(controller.stop());
                    }
```
controller.stop(); obviously does not work and I need help, thanks!!

**Comment by amenbros** [comment, marked as answer]
I've figured it out by doing
```
let stopCollector = collector.messagesCollector
stopCollector.stop()
```

**Reply by IDjinn** [reply]
Sorry for dont reply, i didn't found this topic.

Yes, if you use `controller.stop()` it will stop reactions and messages collector, use `controller.messagesCollector.stop()` to just stop message collector
